src/ledger/LedgerManagerImpl.cpp

```diff
@@ -536,6 +536,10 @@ LedgerManagerImpl::finalizeCatchup(LedgerCloseData const& ledgerData)
 void
 LedgerManagerImpl::addToSyncingLedgers(LedgerCloseData const& ledgerData)
 {
+    assert(mState == LM_CATCHING_UP_STATE);
```

Comment:
not a fan of `asserts` like this:
* as mentioned in the issue, asserts without logs are a pain to debug
* `addToSyncingLedgers` is a helper function, ideally it should not have a dependency on the state machine as it makes the code harder to reason about (if there is a dependency - the `SyncingLedgerChain` abstraction should be improved)
* can't we just push responsibility of performing the right checks upstream in the call stack?
* if we want it to care about state, it should whitelist instead of blacklist

Reply:
I use such asserts as preconditions - adding logging here would serve no purpose, also moving asserts up the chan would remove the precondtions aspects from these asserts.

I would love to improve abstractions here, I thought I made a issue for that, but I cannot find it, so I probably didn't.

I think we should rework catchup/ledger abstractions a bit to make 100% separation between catchup code and applying current ledger code. That would probably require making another class that would decide whether to send given ledger to catchup or ledger class. But this is too big to implement here.